packages/react/src/components/TextField/TextField.tsx

```diff
@@ -72,7 +72,8 @@ const PasswordField: ForwardRefExoticComponent<TextFieldProps> = forwardRef(
         ref={ref}
         type={showPassword ? 'text' : 'password'}
         InputProps={{
-          endAdornment: (
+          ...InputProps,
+          endAdornment: InputProps.endAdornment || (
```

Comment:
I'm thinking whether it's really useful to let end users pass adornment because the password masking/unmasking logic in PasswordField component is tightly coupled to the default endAdornment.

Reply:
Ack